### PR TITLE
new constructor for PathTrie 

### PR DESCRIPTION
**Pull request by beiwei30**
introduce new constructor so that PathTrie can use other segmenter than the default PathSegmenter

### CODE REVIEW

File: path_trie.go [hunk 1]
```diff
@@ -21,6 +21,14 @@ func NewPathTrie() *PathTrie {
 	}
 }
 
+// NewPathTrieWithSegmenter allocates and returns a new *PathTrie with the given StringSegmenter implementation
+// other than PathSegmenter.
+func NewPathTrieWithSegmenter(segmenter StringSegmenter) *PathTrie {
```
Comment (dghubble) [inline path_trie.go:26]: I'm fine with `func NewPathTrie` accepting a `type PathTrieConfig struct` with a single `StringSegmenter` field. This library doesn't do releases, so those using it via modules will be picking a sha and those opting to use latest can update or pin. Passing nil should retain the same default though.

Comment (beiwei30) [reply path_trie.go:26]: sure, thanks :) I believe other folks may benefit from this extension mechanism too.

Comment (beiwei30) [reply path_trie.go:26]: looks like today's GitHub fails to pick up the follow up fix. Would you mind to review further on pull request #16?

**Comment by beiwei30** [comment]
looks like today's GitHub fails to pick up the follow up fix. Would you mind to review further on pull request #16?

**Pull request closed without merging**